Sources/ClientRuntime/Networking/BaseError/BaseError.swift

```diff
@@ -0,0 +1,17 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+public protocol BaseError {
+    var httpResponse: HttpResponse { get }
+    var code: String { get }
+    var message: String? { get }
+    var requestID: String? { get }
+}
```

Comment:
This protocol provides a way to expose info common to all service errors.  Each AWS protocol provides a custom type that conforms to this protocol.  There is also a "test" base error used by the WeatherSDK.